components/script/dom/trackevent.rs

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::codegen::Bindings::EventBinding::EventBinding::EventMethods;
+use crate::dom::bindings::codegen::Bindings::TrackEventBinding;
+use crate::dom::bindings::codegen::Bindings::TrackEventBinding::TrackEventMethods;
+use crate::dom::bindings::codegen::UnionTypes::VideoTrackOrAudioTrackOrTextTrack;
+use crate::dom::bindings::error::Fallible;
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject};
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::bindings::str::DOMString;
+use crate::dom::event::Event;
+use crate::dom::globalscope::GlobalScope;
+use crate::dom::window::Window;
+use dom_struct::dom_struct;
+use servo_atoms::Atom;
+
+#[dom_struct]
+pub struct TrackEvent {
+    event: Event,
+    #[ignore_malloc_size_of = "Defined in rust-mozjs"]
+    track: Option<VideoTrackOrAudioTrackOrTextTrack>,
+}
+
+impl TrackEvent {
+    pub fn new_inherited(_track: &Option<VideoTrackOrAudioTrackOrTextTrack>) -> TrackEvent {
+        unimplemented!();
```

Comment:
We can match on the input enum and call `Dom::from_ref` to store in a new instance of the enum I suggested adding.